### PR TITLE
Improve Custom Format compare logging

### PR DESCRIPTION
**Pull request by bakerboy448**
#### Database Migration
NO

#### Description
title

- radarr downstream: https://github.com/Radarr/Radarr/commit/4bcde25e2916ef7727a2f2d7e481501653acc856
- radarr downstream: https://github.com/Radarr/Radarr/pull/9501

#### Todos
- Tests
- Wiki Updates

#### Issues Fixed or Closed by this PR

*

### CODE REVIEW

**Comment by thezak48** [comment]
`New file's custom formats [{0}] score of [{1}] do not improve on [{2}] score of ({3}), skipping
`
That feels easier to say

**Comment by stevietv** [comment]
To be consistent with the parenthesis or square brackets

> `New file's custom formats [{0}] score of ({1}) does not improve on [{2}] score of ({3}), skipping ` That feels easier to say

**Comment by bakerboy448** [comment]
> To be consistent with the parenthesis or square brackets
> 
> > `New file's custom formats [{0}] score of ({1}) does not improve on [{2}] score of ({3}), skipping `

need to differentiate the CFs vs the score.

Aligned with downstream and added logging for both the decision engine and import

File: src/NzbDrone.Core/DecisionEngine/Specifications/UpgradableSpecification.cs [hunk 1]
```diff
@@ -78,7 +78,11 @@ public bool IsUpgradable(QualityProfile qualityProfile, QualityModel currentQual
                 return false;
             }
 
-            _logger.Debug("New item has a better custom format score");
+            _logger.Debug("New item's custom formats [{0}] ({1}) do improve on [{2}] ({3}), accepting",
```
Comment (stevietv) [inline src/NzbDrone.Core/DecisionEngine/Specifications/UpgradableSpecification.cs:81]:
```suggestion
            _logger.Debug("New item's custom formats [{0}] ({1}) improve on [{2}] ({3}), accepting",
```

File: src/NzbDrone.Core/MediaFiles/EpisodeImport/Specifications/UpgradeSpecification.cs [hunk 2]
```diff
@@ -60,16 +60,26 @@ public Decision IsSatisfiedBy(LocalEpisode localEpisode, DownloadClientItem down
...
+                    _logger.Debug("New item's custom formats [{0}] ({1}) do not improve on [{2}] ({3}), skipping",
+                        newFormats.ConcatToString(),
+                        newFormatScore,
+                        currentFormats.ConcatToString(),
+                        currentFormatScore);
 
                     return Decision.Reject("Not a Custom Format upgrade for existing episode file(s)");
                 }
+
+                _logger.Debug("New item's custom formats [{0}] ({1}) do improve on [{2}] ({3}), accepting",
```
Comment (stevietv) [inline src/NzbDrone.Core/MediaFiles/EpisodeImport/Specifications/UpgradeSpecification.cs:78]:
```suggestion
                _logger.Debug("New item's custom formats [{0}] ({1}) improve on [{2}] ({3}), accepting",
```

**Review by stevietv — CHANGES_REQUESTED** [review]
the word 'do' is a bit superfluous.

File: src/NzbDrone.Core/MediaFiles/EpisodeImport/Specifications/UpgradeSpecification.cs [hunk 3]
```diff
@@ -60,16 +60,26 @@ public Decision IsSatisfiedBy(LocalEpisode localEpisode, DownloadClientItem down
...
+                if (qualityCompare == 0 && newFormatScore < currentFormatScore)
                 {
-                    _logger.Debug("New file's custom formats [{0}] do not improve on [{1}], skipping",
-                        localEpisode.CustomFormats.ConcatToString(),
-                        currentFormats.ConcatToString());
+                    _logger.Debug("New item's custom formats [{0}] ({1}) do not improve on [{2}] ({3}), skipping",
+                        newFormats.ConcatToString(),
+                        newFormatScore,
+                        currentFormats.ConcatToString(),
+                        currentFormatScore);
```
Comment (stevietv) [inline src/NzbDrone.Core/MediaFiles/EpisodeImport/Specifications/UpgradeSpecification.cs:73]:
````suggestion
                        newFormats != null ? newFormats.ConcatToString() : "",
                        newFormatScore,
                        currentFormats != null ? currentFormats.ConcatToString() : "",
                        currentFormatScore);
````

File: src/NzbDrone.Core/MediaFiles/EpisodeImport/Specifications/UpgradeSpecification.cs [hunk 4]
```diff
@@ -60,16 +60,26 @@ public Decision IsSatisfiedBy(LocalEpisode localEpisode, DownloadClientItem down
...
 
                     return Decision.Reject("Not a Custom Format upgrade for existing episode file(s)");
                 }
+
+                _logger.Debug("New item's custom formats [{0}] ({1}) do improve on [{2}] ({3}), accepting",
+                    newFormats.ConcatToString(),
+                    newFormatScore,
+                    currentFormats.ConcatToString(),
+                    currentFormatScore);
             }
```
Comment (stevietv) [inline src/NzbDrone.Core/MediaFiles/EpisodeImport/Specifications/UpgradeSpecification.cs:83]:
```suggestion
                        newFormats != null ? newFormats.ConcatToString() : "",
                        newFormatScore,
                        currentFormats != null ? currentFormats.ConcatToString() : "",
                        currentFormatScore);
```

**Review by stevietv — CHANGES_REQUESTED** [review]

**Comment by stevietv** [comment]
the null check will fix the failing tests in the build.

**Comment by bakerboy448** [comment]
Preciate the null fixups! :)

**Review by markus101 — APPROVED** [review]

**Pull request merged**